deltacat/compute/compactor/model/delta_file_envelope.py

```diff
@@ -63,9 +73,14 @@ def file_index(self) -> int:
     def delta_type(self) -> DeltaType:
         return DeltaType(self["deltaType"])
 
+    @property
+    def table_storage_strategy(self) -> LocalTableStorageStrategy:
+        return self["tableStorageStrategy"]
+
     @property
     def table(self) -> LocalTable:
-        return self["table"]
```

Comment:
can this method handle ray.get and making table a Union?

Reply:
I don't follow - the implementation of LocalTableStorageStrategy handles any underlying calls to RayPlasmaObjectStore (if that's what it implements)

The return type of the LocalTableNOOPStorageStrategy and LocalTableRayObjectStoreReferenceStorageStrategy which implement LocalTableStorageStrategy is `LocalTable`

```
from deltacat.storage import (
    LocalTable,
)

LocalTableReference = Union[ObjectRef, LocalTable]


class LocalTableStorageStrategy(ABC):
    @abstractmethod
    def store_table(self, table: LocalTable) -> LocalTableReference:
        pass

    @abstractmethod
    def get_table(self, table_like: LocalTableReference) -> LocalTable:
        pass


class LocalTableNOOPStorageStrategy(LocalTableStorageStrategy):
    """
    Stores the table as is without any modifications
    """

    def store_table(self, table: LocalTable) -> LocalTableReference:
        return table

    def get_table(self, table_like: LocalTableReference) -> LocalTable:
        return table_like


class LocalTableRayObjectStoreReferenceStorageStrategy(LocalTableStorageStrategy):
    """
    Stores the table in the RayPlasmaObjectStore - see deltacat/io/ray_plasma_object_store.py
    """

    def store_table(self, table: LocalTable) -> LocalTableReference:
        obj_ref: ObjectRef = RayPlasmaObjectStore().put(table)
        return obj_ref

    def get_table(self, table_like: LocalTableReference) -> LocalTable:
        table = RayPlasmaObjectStore().get(table_like)
        return table

```

Reply:
Yeah, i am saying `table` can represent `Union[ObjectRef[LocalTable], LocalTable]` and this property calls ray.get if table is `ObjectRef[LocalTable]`.